addons/web_editor/static/src/js/wysiwyg_snippets/wysiwyg_snippets.js

```diff
@@ -71,9 +71,17 @@ Wysiwyg.include({
                 $editable.before($wrap);
                 $wrap.append($editable);
                 setTimeout(function () { // add a set timeout for the transition
-                    self.snippets.$el.addClass('o_loaded');
-                    self.$el.addClass('o_snippets_loaded');
-                    self.trigger_up('snippets_loaded', self.snippets.$el);
+                    var totalDisabledSnippets = 0;
+                    _.each(self.snippets.$snippets, function (el) {
+                        if ($(el).hasClass('o_disabled')) {
+                            totalDisabledSnippets += 1;
+                        }
+                    });
+                    if (self.snippets.$snippets.length !== totalDisabledSnippets) {
+                        self.snippets.$el.addClass('o_loaded');
+                        self.$el.addClass('o_snippets_loaded');
+                        self.trigger_up('snippets_loaded', self.snippets.$el);
+                    }
```

Comment:
@qsm-odoo 
In reply to @JKE-be, this is the code to show snippets bar and hide it if snippets are all disabled (in red).
Is this ok or do you have maybe a better workaround to check if all snippets are disabled ? Thanks

Reply:
Not sure since this is the new editor's code but was there a problem to rather do something where the 'o_disabled' class is added ? I feel it would make more sense... but I don't know, did you analyze the code you are wrapping in a `if` ? I feel that all that "loaded" stuff should still be done no matter if snippets are droppable or not. Also, you condition seems to be simply `self.$snippets.$snippets.not('.o_disabled').length` (no each and counting and ...)...

I feel that what should be done is toggling a new class on the snippet panel depending on if there are droppable snippets or not... and so this should be done where we determine if snippets are droppable (where we add the `o_disabled` class).

Reply:
Thanks ! Indeed, this condition is much simpler... :hear_no_evil: 
Well, what shows the snippets bar is the `o_loaded` and `o_snippets_loaded` classes.

Reply:
Yes, but that does not change what I explained. The fact that the snippets bar is "loaded" has nothing to do with the fact that it contains disabled snippets or not (note that it is probably bad that the o_loaded class is added there... but well that is the new editor :cry:).

I think the correct behavior in the current implementation would be:
1. Create the snippets bar
2. Disable / Enable snippets
3. **Check if all are disabled, in that case, add a new class on the panel**
4. Add the o_loaded class as it is now loaded

Implementing it this way seems more logical and would take care of the fact, that snippets might become disabled during the edition (each time we recheck if we have to add the disabled class or not). The condition for the snippets bar to be shown would be to have the `o_loaded` class **and** *not* the new class you may have toggled at step 3.

Reply:
@qsm-odoo I'm blocking at the last step.
In fact I added `this.$el.toggleClass('o_all_snippets_disabled', this.$snippets.not('.o_disabled').length === 0);` in the `_disableUndroppableSnippets` function. So, I can have `o_all_snippets_disabled` class when all snippets are disabled. Here is where I block : 

> The condition for the snippets bar to be shown would be to have the o_loaded class and not the new class you may have toggled at step 3.

Yes, indeed that is the condition I need but where should I put this ? What confuses me is the `o_loaded` and `o_snippets_loaded` classes which display the snippets bar. I also tried with something like `:not(.o_all_snippets_disabled)` in scss.

Reply:
Have a look at the last commit, at least it is working like that but I guess this is not what you expected ?

Reply:
Mmh, it is because the scss is a mess. I'll check but could you extract from your two commits the part which concerns all of this into a single commit (so just one commit for the snippets panel and one commit for the rest of your work) ? Thanks.